src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace AppBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * This listener subscribes to the 'kernel.request' event to redirect a client
+ * to an appropriate locale, according to browser settings.
+ *
+ * See http://symfony.com/doc/current/components/http_kernel/introduction.html#the-kernel-request-event
+ *
+ * @author Oleg Voronkovich <oleg-voronkovich@yandex.ru>
+ */
+class RedirectToPreferredLocaleListener
+{
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    /**
+     * List of supported locales.
+     *
+     * @var string[]
```

Comment:
not needed. the phpdoc standard already says that ``string[]`` is an array of strings